docs/asf-docs/Introduction_hyp3_mintpy.ipynb

```diff
@@ -0,0 +1,510 @@
+   "source": [
+    "User autentication"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hyp3 = sdk.HyP3(prompt = True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Submit the jobs with hyp3-sdk and download the files "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "reference_granule = 'S1A_IW_SLC__1SSV_20160628T205108_20160628T205136_011916_01259C_EE11'\n",
+    "\n",
+    "secondary_granules = [\n",
+    "    'S1A_IW_SLC__1SSV_20160112T205113_20160112T205140_009466_00DBA8_C430',\n",
+    "    'S1A_IW_SLC__1SSV_20160205T205112_20160205T205139_009816_00E5D1_997E',\n",
+    "    'S1A_IW_SLC__1SSV_20160229T205112_20160229T205139_010166_00EFF4_357E',\n",
+    "    'S1A_IW_SLC__1SSV_20160324T205112_20160324T205139_010516_00F9E8_FA1C',\n",
+    "    'S1A_IW_SLC__1SSV_20160417T205113_20160417T205140_010866_010443_7429',\n",
+    "    'S1A_IW_SLC__1SSV_20160511T205115_20160511T205142_011216_010F42_BD06',\n",
+    "    'S1A_IW_SLC__1SSV_20160604T205107_20160604T205135_011566_011A9C_EDD4',\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "insar_jobs = sdk.Batch()\n",
+    "\n",
+    "for secondary_granule in secondary_granules:\n",
+    "    insar_jobs += hyp3.submit_insar_job(reference_granule, secondary_granule,\n",
+    "                                        job_name=projectname, include_inc_map=True, include_dem=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "insar_jobs = hyp3.watch(insar_jobs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "insar_jobs.download_files()\n",
+    "print(\"completed ...\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Organize the data into the structure that MintPy requests"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### unzip zip files in the current directory"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_sublist_via_pattern(lst, pattern_str):\n",
+    "    pattern = re.compile(pattern_str)\n",
+    "    return list(filter(pattern.search, lst))\n",
+    "\n",
+    "# unzip zip files in current directory\n",
+    "filelist = glob.glob(\"{}/*.zip\".format(outdir))\n",
+    "# split the filelist into two lists\n",
+    "p1 = \"/[A-Z][0-9][A-Z][A-Z]_\"\n",
+    "filelist1 = get_sublist_via_pattern(filelist, p1)\n",
+    "for file in filelist1:\n",
+    "    with zipfile.ZipFile(file, \"r\") as zip_ref:\n",
+    "        zip_ref.extractall(outdir)"
```

Comment:
the `download_files()` function returns a list of `Path` objects to the product zips, so download + unzip can be reduced to:
```
zip_files = insar_jobs.download_files()

for zip_file in zip_files:
    with zipfile.ZipFile(zip_file) as zip_ref:
        zip_ref.extractall(outdir)
```

Reply:
changed